### PR TITLE
Added implementation of UniqueIngredientList

### LINKED ISSUE #76: Added UniqueIngredientList

### PR DESCRIPTION
**Pull request by tzihiang**
Created `ingredient` folder, and inside it, added `exceptions`. This is mainly to facilitate the `UniqueIngredientList`.

In `ingredient.exception`, added `DuplicateIngredientException` and `IngredientNotFoundException`.

Under `model`, added `ReadOnlyIngredientList` and `Inventory`.

### CODE REVIEW

**Review by shanecsj — APPROVED** [review]
Need to change all the 'address book' in the comments to 'inventory'. 
Other than that, LGTM.

**Review by hans555 — APPROVED** [review]
Looks good

File: src/main/java/seedu/address/model/Inventory.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    /**
+     * Replaces the contents of the ingredient list with {@code ingredients}.
+     * {@code ingredients} must not contain duplicate ingredients.
+     */
+    public void setIngredients(List<Ingredient> ingredients) {
+        this.ingredients.setIngredients(ingredients);
+    }
+
+
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/model/Inventory.java]: Extra line

File: src/main/java/seedu/address/model/ReadOnlyIngredientList.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package seedu.address.model;
+
+import javafx.collections.ObservableList;
+import seedu.address.model.ingredient.Ingredient;
+/**
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/model/ReadOnlyIngredientList.java:6]: New line after last import

File: src/main/java/seedu/address/model/ingredient/UniqueIngredientList.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        internalList.add(toAdd);
+    }
+
+    public void setIngredients(UniqueIngredientList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code persons}.
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/model/ingredient/UniqueIngredientList.java]: The javadocs comments mention person instead of ingredient

**Review by teo-jun-xiong — APPROVED** [review]
LGTM

**Review by gohkh — APPROVED** [review]

**Pull request merged**